lib/game.rb

```diff
@@ -0,0 +1,73 @@
+class Game
+
+  attr_accessor :board, :player_1, :player_2
+
+  WIN_COMBINATIONS = [
+    [0,1,2],
+    [3,4,5],
+    [6,7,8],
+    [0,3,6],
+    [1,4,7],
+    [2,5,8],
+    [0,4,8],
+    [2,4,6]
+    ]
+
+  def initialize(player_1 = Players::Human.new("X"), player_2 = Players::Human.new("O"), board = Board.new)
+   @player_1 = player_1
+   @player_2 = player_2
+   @board = board
+   # binding.pry
+   end
+
+   def current_player
+     board.turn_count % 2 == 0 ? self.player_1 : self.player_2
+   end
+
+   def over?
+     self.draw? || self.won?
+   end
+
+   def won?
+     WIN_COMBINATIONS.detect do |a|
+       @board.cells[a[0]] == @board.cells[a[1]] &&
+       @board.cells[a[0]] == @board.cells[a[2]] &&
+       @board.taken?(a[0]+1)
+     end
+   end
+
+   def draw?
+     self.board.turn_count == 9 && !self.won?
+   end
+
+   def winner
+     self.won? ? self.board.cells[self.won?[0]]:nil
```

Comment:
Instead of calling `Game#won?` twice, consider using an if statement assignment. For example:

```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.